src/components/block/block.ts

```diff
@@ -0,0 +1,320 @@
+import EventBus from '../../utils/eventBus';
+import { compile as pugCompile, compileTemplate as TCompileTemplate } from 'pug';
+import { v4 as uuid } from 'uuid';
+
+export type TChild = {
+	key: string
+	value: Block
+	constructor: TComponentConstructor<TProps, Block>
+	props: TProps
+};
+
+export type TCallback = (...args: unknown[]) => void
+
+export type TListener = {
+	selector: string
+	event: string
+	callback: TCallback
```

Comment:
Если используешь в проекте `;`, то лучше использовать везде для единообразия
```suggestion
export type TCallback = (...args: unknown[]) => void;

export type TListener = {
	selector: string;
	event: string;
	callback: TCallback;
```